pkg/gardenlet/controller/seed/seed_control.go

```diff
@@ -176,6 +177,21 @@ func (c *defaultControl) ReconcileSeed(obj *gardencorev1beta1.Seed, key string)
 		return fmt.Errorf("failed to get garden client: %w", err)
 	}
 
+	// Initialize capacity and allocatable
+	var capacity, allocatable corev1.ResourceList
+	if c.config.Resources != nil && len(c.config.Resources.Capacity) > 0 {
+		capacity = make(corev1.ResourceList, len(c.config.Resources.Capacity))
+		allocatable = make(corev1.ResourceList, len(c.config.Resources.Capacity))
+		for resourceName, quantity := range c.config.Resources.Capacity {
+			capacity[resourceName] = quantity
+			if reservedQuantity, ok := c.config.Resources.Reserved[resourceName]; ok {
+				allocatable[resourceName] = *resource.NewQuantity(quantity.Value()-reservedQuantity.Value(), quantity.Format)
```

Comment:
Can you use `Quantity.Sub` here to be on the safe side? Might not be relevant for integers like the number of shoots, but I would like to stick to the idiomatic use of `resource.Quantity`.
```suggestion
				allocatableQuantity := quantity.DeepCopy()
				allocatableQuantity.Sub(reservedQuantity)
				allocatable[resourceName] = allocatableQuantity
```